infer/src/bufferoverrun/bounds.ml

```diff
@@ -1136,8 +1143,7 @@ module Bound = struct
           in
           match get s bound_position with
           | Bottom ->
-              Option.value_map (big_int_of_minmax subst_pos x) ~default:Bottom ~f:(fun i ->
-                  NonBottom (of_big_int i) )
+              overapproximate_bound bound_position (Symb.Symbol.is_unsigned s)
```

Comment:
Thanks for catching this. Yes, this is indeed better. We could also use `overapproximate_bound` to get the default:

```
                (* Pass false as unsigned to overapproximate_bound because the bound doesn't
                   consist only of the symbol s so it cannot be overapproximated with 0 even
                   if s is signed and subst_pos is LowerBound. *)
                Option.value_map (big_int_of_minmax subst_pos x)
                  ~default:(overapproximate_bound subst_pos false) ~f:(fun i ->
                    NonBottom (of_big_int i) )
```